src/script/conversation/ConversationRepository.ts

```diff
@@ -593,29 +593,27 @@ export class ConversationRepository {
           failedToAddUsers,
           conversationEntity,
           this.userState.self().id,
+          AddUsersFailureReasons.UNREACHABLE_BACKENDS,
         );
         await this.eventRepository.injectEvent(failedToAddUsersEvent);
       }
 
       return conversationEntity;
     } catch (error) {
-      if (!isBackendError(error)) {
-        this.logger.error(error);
-        throw error;
-      }
-
-      switch (error.label) {
-        case BackendErrorLabel.CLIENT_ERROR:
-          this.handleTooManyMembersError();
-          break;
-        case BackendErrorLabel.NOT_CONNECTED:
-          await this.handleUsersNotConnected(userEntities.map(user => user.qualifiedId));
-          break;
-        case BackendErrorLabel.LEGAL_HOLD_MISSING_CONSENT:
-          this.showLegalHoldConsentError();
-          break;
-        default:
-          this.logger.error(error);
+      if (isBackendError(error)) {
```

Comment:
I actually like this early return patter:
      if (!isBackendError(error)) {
        this.logger.error(error);
        throw error;
        }
        switch ...

Reply:
let's not `logger.log` since we throw, the caller will most likely print the error anyways

Reply:
The problem with early return in error handling is that it might filter too much and makes it a little harder to add new error types to be handled here. 

Another solution, that is easily extensible would be using a switch/case statement

```js
switch (true) {
  case isBackendError(error):
    // handler here

  default
   throw error;
}
```